ignite/metrics/accuracy.py

```diff
@@ -122,12 +121,12 @@ def reset(self):
         super(Accuracy, self).reset()
 
     def update(self, output):
-
-        y_pred, y = self._check_shape(output)
+        y_pred, y = output
+        self._check_shape((y_pred, y))
         self._check_type((y_pred, y))
 
         if self._type == "binary":
-            correct = torch.eq(y_pred.type(y.type()), y).view(-1)
+            correct = torch.eq(y_pred.view(-1).type(y.type()), y.view(-1))
```

Comment:
Maybe we can change the following
```
y_pred.view(-1).type(y.type())
```
by 
```
y_pred.view(-1).to(y)
```
https://pytorch.org/docs/stable/tensors.html#torch.Tensor.to

Reply:
agreed and updated.